external_services/lti.py

```diff
@@ -21,7 +22,29 @@ def __init__(self, service, user, instance, host, title, context_id=None, link_i
         link_id = link_id or "aplus{:d}".format(service.pk)
         title = title or link_id
 
-        student_id = self.external_student_id(user)
+        # Gather user information
+        student_id = "i" + self.external_student_id(user) # i for internal
+        full_name = "{} {}".format(user.first_name, user.last_name)
+        given_name = user.first_name
+        family_name = user.last_name
+        email = user.email
+
+        # Anonymize user information
+        enrollment = Enrollment.objects.filter(course_instance=instance, user_profile=user.userprofile).first()
+        if enrollment:
```

Comment:
tää pitäis varmaan muokata muotoon `if service.is_anonymous:` ja sit sen sisällä `if not enrollment: raise exception`. ja sitten tolle ekalle if:lle else, jossa tehdään toi internal versio.